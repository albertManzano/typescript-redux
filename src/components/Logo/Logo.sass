.Logo
    background-color: white
    padding: 8px
    height: 80%
    box-sizing: border-box
    border-radius: 5px
    &__image
        height: 80%
    &--small
        height: 80px
        padding-left: 15%