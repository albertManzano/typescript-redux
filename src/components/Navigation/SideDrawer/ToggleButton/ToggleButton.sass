.Navigator
    &--line
    width: 90%
    height: 3px
    margin-top: 5px
    background-color: white

@media (max-width: 500px)
    .Navigator--mvl
        display: none

@media (min-width: 499px)
    .Navigator--desktop
        display: none