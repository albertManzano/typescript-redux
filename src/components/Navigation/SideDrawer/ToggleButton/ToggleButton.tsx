import React from 'react'; /* eslint-disable-line */

const ToggleButton = ({ clicked }) => {
	return (
		<div className="Navigator--desktop Navigator__menu" onClick={clicked}>
			<div className="Navigator__menu--line" />
			<div className="Navigator__menu--line" />
			<div className="Navigator__menu--line" />
		</div>
	);
};

export default ToggleButton;
