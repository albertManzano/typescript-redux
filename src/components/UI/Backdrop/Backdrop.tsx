import React from 'react';/* eslint-disable-line */
import './Backdrop.sass';

const Backdrop = ({ show, clickBack }) => (show ? <div className="Backdrop" onClick={clickBack} /> : null);

export default Backdrop;
