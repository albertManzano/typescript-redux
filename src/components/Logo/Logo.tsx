import React from 'react';/* eslint-disable-line */
import burguerLogo from '../../assets/images/burgerLogo.png';

import './Logo.sass';

const Logo = ({size}) => {
	return <>
		{size ? <div className="Logo"><img className="Logo__image" src={burguerLogo} alt="Burguer Logo" /></div>
		: <div className="Logo--small"><img className="Logo__image" src={burguerLogo} alt="Burguer Logo" /></div>}
	</>;
};

export default Logo;
