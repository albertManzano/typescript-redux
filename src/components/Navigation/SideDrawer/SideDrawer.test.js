import React from 'react';

import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import jest from '@testing-library/jest-dom'
import Sidedrawer from './SideDrawer';
import ToggleButton from './ToggleButton/ToggleButton';
import Backdrop from '../../UI/Backdrop/Backdrop';
import Logo from '../../Logo/Logo'
import Aux from '../../../hoc/Aux/Aux'

configure({ adapter: new Adapter() });

describe('<SideDrawer />', () => {
    let wrapper;
    // const clicFn = jest.fn()

    describe('<SideDrawer show/>', () => {
        beforeEach(() => {
            wrapper = shallow(<Sidedrawer show={true} />)
        });

        it('should exist <div class="close">', () => {
            expect(wrapper.html('div.SideDrower')).toContain('open')
        });

        it('should render open drawer', () => {
            expect(wrapper).toMatchSnapshot()
        });

        it('should render the logo and NavigationItems if show is true', () => {
            expect(wrapper.find(<Logo />)).toBeDefined
        });
    });

    describe('<SideDrawer show={false}/>', () => {
        beforeEach(() => {
            wrapper = shallow(<Sidedrawer show={false} />)
        });

        it('should exist <div class="close">', () => {
            expect(wrapper.html('div.SideDrower')).toContain('close')
        });

        it('should render close drawer', () => {
            expect(wrapper).toMatchSnapshot()
        });

    }); 
    
}); 