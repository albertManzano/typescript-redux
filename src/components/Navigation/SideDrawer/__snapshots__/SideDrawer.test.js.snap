// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer /> <SideDrawer show/> should render open drawer 1`] = `ShallowWrapper {}`;

exports[`<SideDrawer /> <SideDrawer show={false}/> should render close drawer 1`] = `ShallowWrapper {}`;
